### DISCUSSION TITLE
Route::dispatch does not maintain request parameters, is this intended behaviour?

### DISCUSSION THREAD

**Discussion by j-dexx** [laravel/framework#45332] [Q&A]
We've got a weird case in our app where dispatching a route directly is useful.

```
$newRequest = Request::create($someUrlPath, 'GET', ['a' => 'b']);
return Route::dispatch($newRequest)->getContent();
```

However when it hits the controller of this route it's lost the parameters passed to the request.  Is this intentional?

**Comment by gitmiro** [comment]
I've just stumbled over the same issue.

I debugged this a bit and it seems the [RouteDepdendencyResolverTrait](https://github.com/laravel/framework/blob/7797de80132599971fdaf03f6c0bfd62b5ae15d8/src/Illuminate/Routing/RouteDependencyResolverTrait.php#L85) is not resolving the request with the params provided.

For me this fixed it:

```php
  $response = app()->handle($request); // works
  $response = Route::dispatch($request); // does not work
```

**Reply by memontarius** [reply]
But _app()->handle($request);_ change current request inside laravel. I.e. it is not possible to execute the request in isolation.
